timemachine/fe/topology.py

```diff
@@ -403,30 +403,27 @@ def setup_chiral_restraints(self, restraint_k=1000.0):
             Returns a ChiralAtomRestraint and a ChiralBondRestraint
 
         """
-        chiral_bonds = chiral_utils.find_chiral_bonds(self.mol)
+        mol = self.mol
+        conf = get_romol_conf(mol)
 
-        chiral_atom_restr_idxs = chiral_utils.setup_all_chiral_atom_restr_idxs(self.mol, get_romol_conf(self.mol))
-
-        chiral_atom_params = []
-        for idxs in chiral_atom_restr_idxs:
-            chiral_atom_params.extend(restraint_k for _ in idxs)
-
-        chiral_atom_params = np.array(chiral_atom_params)
-        chiral_atom_restr_idxs = np.array(chiral_atom_restr_idxs)
+        # chiral atoms
+        chiral_atom_restr_idxs = np.array(chiral_utils.setup_all_chiral_atom_restr_idxs(mol, conf))
+        chiral_atom_params = restraint_k * np.ones(len(chiral_atom_restr_idxs))
```

Comment:
note: size of `chiral_atom_params` was adjusted (from `len(chiral_atom_restr_idxs.flatten())` to `len(chiral_atom_restr_idxs)`) -- may need to make a similar adjustment for chiral_bonds

Reply:
Good catch - looks like the previous implementation was just incorrect - (i.e. we had 4 times as many `k` parameters as there were actual # of restraints). I'm guessing the code still ran since we use the same `k`, and every restraint still had a `k` value they could reference.